### PR TITLE
Preparations for relaxed mode work.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2Native.kt [hunk 1]
```diff
@@ -177,6 +177,18 @@ class K2Native : CLICompiler<K2NativeCompilerArguments>() {
...
 
+                put(MEMORY_MODEL, when (arguments.memoryModel) {
+                    "relaxed" -> {
+                        configuration.report(STRONG_WARNING, "Relaxed memory model is not yet functional")
+                        MemoryModel.RELAXED
+                    }
+                    "strict" -> MemoryModel.STRICT
+                    else -> {
+                        configuration.report(ERROR, "Unsupported memory model ${arguments.memoryModel}")
+                        return
```
Comment (SvyatoslavScherbina) [inline backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2Native.kt]: It is more safe to provide some value here (e.g. `MemoryModel.STRICT`) instead of returning

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt [hunk 2]
```diff
@@ -307,12 +307,18 @@ internal class FunctionGenerationContext(val function: LLVMValueRef,
     }
 
     private fun updateReturnRef(value: LLVMValueRef, address: LLVMValueRef) {
-        store(value, address)
+        if (context.memoryModel == MemoryModel.STRICT)
+            store(value, address)
+        else
+            call(context.llvm.updateReturnRefFunction, listOf(address, value))
```
Comment (SvyatoslavScherbina) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt]: Why is this required?

File: backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2Native.kt (hunk 1 above)
Comment (olonho) [reply backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2Native.kt]: Maybe, can include in followup patches.

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt (hunk 2 above)
Comment (olonho) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt]: Just not rely on C++/Kotlin inliner, which not always works.
